sdk/program/src/alt_bn128_compression.rs

```diff
@@ -0,0 +1,493 @@
+            return Ok([0u8; alt_bn128_compression_size::G1_COMPRESSED]);
+        }
+        let g1 = G1::deserialize_with_mode(
+            convert_endianness::<32, 64>(&g1_bytes).as_slice(),
+            Compress::No,
+            Validate::Yes,
+        )
+        .map_err(|_| AltBn128CompressionError::CompressingG1Failed)?;
+        let mut g1_bytes = [0u8; alt_bn128_compression_size::G1_COMPRESSED];
+        G1::serialize_compressed(&g1, g1_bytes.as_mut_slice())
+            .map_err(|_| AltBn128CompressionError::CompressingG2Failed)?;
+        Ok(convert_endianness::<32, 32>(&g1_bytes))
+    }
+
+    pub fn alt_bn128_compression_g2_decompress(
+        g2_bytes: &[u8],
+    ) -> Result<[u8; alt_bn128_compression_size::G2], AltBn128CompressionError> {
+        let g2_bytes: [u8; alt_bn128_compression_size::G2_COMPRESSED] = g2_bytes
+            .try_into()
+            .map_err(|_| AltBn128CompressionError::DecompressingG2Failed)?;
+        if g2_bytes == [0u8; alt_bn128_compression_size::G2_COMPRESSED] {
+            return Ok([0u8; alt_bn128_compression_size::G2]);
+        }
+        let decompressed_g2 =
+            G2::deserialize_compressed(convert_endianness::<64, 64>(&g2_bytes).as_slice())
+                .map_err(|_| AltBn128CompressionError::DecompressingG2Failed)?;
+        let mut decompressed_g2_bytes = [0u8; alt_bn128_compression_size::G2];
+        decompressed_g2
+            .x
+            .serialize_with_mode(&mut decompressed_g2_bytes[..64], Compress::No)
+            .map_err(|_| AltBn128CompressionError::DecompressingG2Failed)?;
+        decompressed_g2
+            .y
+            .serialize_with_mode(&mut decompressed_g2_bytes[64..128], Compress::No)
+            .map_err(|_| AltBn128CompressionError::DecompressingG2Failed)?;
+        Ok(convert_endianness::<64, 128>(&decompressed_g2_bytes))
+    }
+
+    pub fn alt_bn128_compression_g2_compress(
+        g2_bytes: &[u8],
+    ) -> Result<[u8; alt_bn128_compression_size::G2_COMPRESSED], AltBn128CompressionError> {
+        let g2_bytes: [u8; alt_bn128_compression_size::G2] = g2_bytes
+            .try_into()
+            .map_err(|_| AltBn128CompressionError::CompressingG2Failed)?;
+        if g2_bytes == [0u8; alt_bn128_compression_size::G2] {
+            return Ok([0u8; alt_bn128_compression_size::G2_COMPRESSED]);
+        }
+        let g2 = G2::deserialize_with_mode(
+            convert_endianness::<64, 128>(&g2_bytes).as_slice(),
+            Compress::No,
+            Validate::No,
+        )
+        .map_err(|_| AltBn128CompressionError::DecompressingG2Failed)?;
+        let mut g2_bytes = [0u8; alt_bn128_compression_size::G2_COMPRESSED];
+        G2::serialize_compressed(&g2, g2_bytes.as_mut_slice())
+            .map_err(|_| AltBn128CompressionError::CompressingG2Failed)?;
+        Ok(convert_endianness::<64, 64>(&g2_bytes))
+    }
+
+    pub fn convert_endianness<const CHUNK_SIZE: usize, const ARRAY_SIZE: usize>(
```

Comment:
I'm wondering if it would be useful to put in in some util module?

Reply:
hm I don't see an existing util module, not sure where to put it

Reply:
https://github.com/solana-labs/solana/blob/master/sdk/src/program_utils.rs seems like a good candidate.

It has only one function, completely unrelated to endianness or byte processing, but... it's used across `programs/**`, `sdk/programs/**`, so I guess it's good for functions shared across SDK.

Reply:
it's not in sdk/programs I don't have access to this file